### DISCUSSION TITLE
Encoding error occured when I tried to load .sxm, .dat and .3ds files

### DISCUSSION THREAD

**Discussion by Zuotengyuhe** [yuksk/SIDAM#66] [Q&A] [answered]
When I tried to open .sxm files with SIDAM(SIDAM tab -> load Data -> From a file...),  an error message "unknown text encoding" was diplayed (you can see the error message window in a attached photo). 
I used debugger and found that errors occur at a function "ConvertTextEncoding" (as you can see the other attached photo).

 I use Igor9 in my PC(windows11, default language is English).

A colleague who use an English PC also faces the same problem while other colleagues who use PCs whose default language is Japanese do not face the problem.
I do not know whether this information is essential.

Will you help me solve the above problem?
I aspire to use the wonderful tool for analyzing SI-STM data as my colleagues use.

P.S.
I checked below;
(1) I installed SIDAM files in the same way as the way written in README file.
(2) I did not open pxp files saved in the other PCs. That is, I opend files using SIDAM tab -> load Data -> From a file...

<img width="188" alt="Error message" src="https://github.com/user-attachments/assets/7750e5bb-dc5e-48b2-a0c7-d5289a60d2d7">
<img width="419" alt="Error place" src="https://github.com/user-attachments/assets/f919ea05-5421-436d-8c23-4fe6ff6ff151">

**Comment by yuksk** [comment, marked as answer]
## Why the error occurs
The error occurs when both of the following conditions are met:
1. You do not specify the text encoding for Nanonis files in your `SIDAM.toml` file (line 85, `text_encoding`).
1. The default text encoding of Igor Pro is set to "Western". In this case, executing `DefaultTextEncoding; print(TextEncodingName(V_defaultTextEncoding, 0))` in the command window will return "Unknown" in the history area.

When the text encoding is not specified in your `SIDAM.toml` file, SIDAM uses the default text encoding of Igor Pro. However, if the default is "Western," the function `TextEncodingName` returns "Unknown," causing the error.

## How to Solve the Problem
The best solution is to specify the correct text encoding in your `SIDAM.toml` file. If you are unsure of the correct encoding, try using "UTF-8", as it will likely resolve the error. However, if the actual encoding is not "UTF-8" and your Nanonis files contain non-ASCII characters (e.g., in comments or file paths), these characters may become garbled.

Alternatively, you can change the default text encoding of Igor Pro to something other than "Western." While this is an easier solution to try, note that non-ASCII characters might still be garbled, and there could be side effects.

**Comment by Zuotengyuhe** [comment]
Thank you so much!

Both solutions worked!
